### PR TITLE
Add test coverage for the SSTIO class

### PR DESCRIPTION
**Pull request by dannosliwcd**
- Extract the ioctl calls to a mockable object
- Add unit tests for the SSTIO interface

### CODE REVIEW

File: test/SSTIOTest.cpp [hunk 1]
```diff
@@ -0,0 +1,446 @@
...
+    static const uint32_t max_cpus(32);
+    static const uint32_t expected_mbox_read_value(123);
+    static const uint32_t expected_mmio_read_value(456);
+    SSTIOImp sstio(max_cpus, std::static_pointer_cast<SSTIoctl>(m_ioctl));
+
+    // Add a mailbox read, then verify that we can sample it.
+    auto mbox_read_idx = sstio.add_mbox_read(0, 0, 0, 0);
+    {
+        EXPECT_CALL(*m_ioctl, mbox(Field(&sst_mbox_interface_batch_s::num_entries, 1)))
+            .WillOnce([](sst_mbox_interface_batch_s *mbox_batch) {
```
Comment (cmcantalupo) [inline test/SSTIOTest.cpp:230]: Lo, and there was beauty in gmock!

File: test/MockSSTIoctl.hpp [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+#include "gmock/gmock.h"
+
+#include "SSTIoctl.hpp"
+
+class MockSSTIoctl : public geopm::SSTIoctl
+{
+    public:
+        virtual ~MockSSTIoctl() = default;
+
+        MOCK_METHOD1(version, int(geopm::sst_version_s *version));
```
Comment (dannosliwcd) [inline test/MockSSTIoctl.hpp]: See if we can use new-style MOCK_METHOD calls

Comment (dannosliwcd) [reply test/MockSSTIoctl.hpp]: also use override qualifiers

**Review by cmcantalupo — DISMISSED** [review]
Great work!

**Review by cmcantalupo — APPROVED** [review]

**Pull request merged**